### PR TITLE
DBZ-6592 Documenting database permissions required by the MongoDB connector

### PR DESCRIPTION
**Pull request by jcechace**
https://issues.redhat.com/browse/DBZ-6592

### CODE REVIEW

**Comment by jcechace** [comment]
@roldanbob  as usual, please review and suggest of directly commit any changes as you see fit. Thank you!

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 1]
```diff
@@ -181,6 +181,18 @@ MongoDB does not recommend running a standalone server in production.
...
 ====
 
+// Type: concept
+// ModuleID: mongodb-user-permissions-required-by-debezium-connectors
+// Title: User permissions required by {prodname} connectors
+[[required-user-permissions]]
+=== Required user permissions
+Depending on the value of xref:mongodb-property-capture-scope[`capture.scope`] property the user used by the connector needs either the permissions to read any database (when xref:mongodb-property-capture-scope[`capture.scope`] is set to `deployment`) or the permissions to read the database specified by xref:mongodb-property-capture-target[`capture.target`] (when xref:mongodb-property-capture-scope[`capture.scope`] is set to `database`).
+
+In any case the user should also have the permission to execute the https://www.mongodb.com/docs/manual/reference/command/ping/[ping] command.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:191]:
```suggestion
.Permission to use the MongoDB `ping` command 

Regardless of the `capture.scope` setting, the user requires permission to run the MongoDB https://www.mongodb.com/docs/manual/reference/command/ping/[ping] command.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 2]
```diff
@@ -181,6 +181,18 @@ MongoDB does not recommend running a standalone server in production.
...
+
+Additionally when capturing changes from a sharded MongoDB cluster and the xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] property is set to `replica_set` the user used by the connector also needs the permission to read the `config.shards` system collection.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:193]:
```suggestion
.Permission to read the `config.shards` collection

For connectors that capture changes from a sharded MongoDB cluster, and for which the xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] property is set to `replica_set`, you must grant the user permission to read the `config.shards` system collection.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 3]
```diff
@@ -1518,6 +1530,24 @@ This property does not affect the behavior of incremental snapshots. +
 To match the name of a schema, {prodname} applies the regular expression that you specify as an _anchored_ regular expression.
 That is, the specified expression is matched against the entire name string of the schema; it does not match substrings that might be present in a schema name.
 
+|[[mongodb-property-capture-scope]]<<mongodb-property-capture-scope, `+capture.scope+`>>
+|`deployment`
+|Specifies the https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment[scope of change streamed] opened by the connector.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1535]:
```suggestion
|Specifies the https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment[scope of the change streams] that the connector opens.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 4]
```diff
@@ -1518,6 +1530,24 @@ This property does not affect the behavior of incremental snapshots. +
...
+Set this property to one of the following values:
+
+`deployment`:: opens a change stream cursor for a deployment (either a replica set or a sharded cluster) to watch for changes to all non-system collections across all databases except for `admin`, `local`, and `config`.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1538]:
```suggestion
`deployment`:: Opens a change stream cursor for a deployment (either a replica set or a sharded cluster) to watch for changes to all non-system collections across all databases, except for `admin`, `local`, and `config`.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 5]
```diff
@@ -1518,6 +1530,24 @@ This property does not affect the behavior of incremental snapshots. +
...
+
+`database`:: opens a change stream cursor for a single database to watch for changes to all its non-system collections.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1540]:
```suggestion
`database`:: Opens a change stream cursor for a single database to watch for changes to all of its non-system collections.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 6]
```diff
@@ -1518,6 +1530,24 @@ This property does not affect the behavior of incremental snapshots. +
...
++
+[WARNING]
+====
+To use signaling when `database` is specified as the value the signaling collection needs to be located in the database specified by the xref:mongodb-property-capture-target[`capture.target`] property.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1544]:
```suggestion
If you set the value of this property to `database`,  to use {prodname} signaling , the signaling collection must reside in the database that is  specified by the xref:mongodb-property-capture-target[`capture.target`] property.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 7]
```diff
@@ -1518,6 +1530,24 @@ This property does not affect the behavior of incremental snapshots. +
...
+====
+
+|[[mongodb-property-capture-target]]<<mongodb-property-capture-target, `+capture.target+`>>
+|
+| Specifies the database to watch for changes when xref:mongodb-property-capture-scope[`capture.scope`] is set to `database`
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1549]:
```suggestion
| Specifies the database that the connector monitors for changes.
This property applies only if the xref:mongodb-property-capture-scope[`capture.scope`] is set to `database`.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
Here you go, @jcechace. Let me know if you have any questions.
Thanks!

**Comment by jcechace** [comment]
@roldanbob  Great, this is much better (as is usually the case after your touch). Thank you!

**Comment by jcechace** [comment]
@roldanbob  reread please and accept the review if this can be merged.

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 8]
```diff
@@ -1518,6 +1554,25 @@ This property does not affect the behavior of incremental snapshots. +
...
+|Specifies the https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment[scope of the change streams] that the connector opens.
+Set this property to one of the following values:
+
+`deployment`:: Opens a change stream cursor for a deployment (either a replica set or a sharded cluster) to watch for changes to all non-system collections across all databases, except for `admin`, `local`, and `config`.
+
+`database`:: Opens a change stream cursor for a single database to watch for changes to all of its non-system collections.
++
+[WARNING]
+====
+If you set the value of this property to `database`,  to use {prodname} signaling , the signaling collection must reside in the database that is  specified by the xref:mongodb-property-capture-target[`capture.target`] property.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1568]:
```suggestion
If you set the value of this property to `database`, to use {prodname} signaling, you the signaling collection that you create must reside in the database that is specified by the xref:mongodb-property-capture-target[`capture.target`] property.
```
Removed some extra spaces and slightly reworded.

**Review by roldanbob — CHANGES_REQUESTED** [review]
@jcechace Review complete. Found a couple things that I missed earlier. If you could work in these changes, I think this is now all set. Thanks!

**Review by roldanbob — APPROVED** [review]
LGTM!

**Pull request merged**